### PR TITLE
Add Dockerfile and update README.

### PR DESCRIPTION
**Pull request by danielballan**
I could do a better job spelling out on the environment variables, but I expect that
to be easier once the Helm chart is written (which has to consolidate all this stuff
anyway).

### CODE REVIEW

**Comment by Mr0grog** [comment]
It has a slightly more developer-y focus, but I’m going to close #16 as superseded by this. LMK if you want to keep it around @danielballan.

**Comment by Mr0grog** [comment]
Hmmmm, not sure really affects this PR, but worth keeping in mind that this build process will include a local `.env` file in the image if you’ve got one. So if we’re building an image we plan to push to a public registry (e.g. docker hub), we should be careful about that.

File: README.md [hunk 1]
```diff
@@ -163,6 +163,17 @@ If you are getting errors such as `FATAL: role "user" doesn't exist. Couldn't cr
...
+
+The Dockerfile runs the rails server on port 80 in the container. To build
+and run:
+
+```
+docker build -t db . -e <ENVIRONMENT VARIABLES>
+docker run -p 3000:3000 db
+```
+
+Point your browser or ``curl`` at ``http://localhost:4000``.
```
[the remark this answers is not in the document]
Comment (danielballan) [reply README.md]: Right. Leaked from copy/paste.

**Comment by danielballan** [comment]
Good point! I just checked all my build contexts carefully, and fortunately nothing has been leaked. (My local `.env` files are all straight copies of `.env.example` -- no real keys or login credenitials.) I discovered `.dockerignore`, which is designed to address this problem, and added it to all three PRs.

```
#.dockerignore
.env
.git/
```

Before:

```
# ls .git
COMMIT_EDITMSG	HEAD	   config	hooks  info  objects	  refs
FETCH_HEAD	ORIG_HEAD  description	index  logs  packed-refs
```

After:

```
# ls .git
ls: cannot access .git: No such file or directory
```

and likewise for `.env`.

**Comment by danielballan** [comment]
> Are you planning to add configurations for a worker image in this PR (either in this Dockerfile or a separate one)?

Yes, we might as well wait to merge all these Docker PRs until https://github.com/edgi-govdata-archiving/web-monitoring/issues/73 is complete. I plan to handle Redis in a separate Dockerfile in the same PR.

**Comment by danielballan** [comment]
(Need to update README to account for import-worker.)

**Comment by Mr0grog** [comment]
Hmmm, with that last one, I feel like having each `Dockerfile` in its own directory is a little painful. Why not have either:

- A single, multi-stage Dockerfile, or
- A `docker` directory like you’ve got, but instead of having subdirectories, just have `Dockerfile`, `Dockerfile.server`, and `Dockerfile.worker`?

**Comment by danielballan** [comment]
Looks like support for file extensions was [added since the last time I checked](https://stackoverflow.com/a/32108873/1221924). I was Googling for something like multi-stage Dockerfile and failed to discover it. That seems like the better option, but I'll need to take some time to understand it.

**Comment by danielballan** [comment]
In the commit fa08c43 there is a working Dockerfile. In commit a263f04 I un-comment some lines that turn it into a multi-stage build. That changes causes this error during the build:

```
LoadError: libpq.so.5: cannot open shared object file: No such file or directory - 
```
Open details for context:

<details>
```
$ docker build -f Dockerfile.rails-server .
Sending build context to Docker daemon  92.49MB
<snip>
Step 12/16 : COPY --from=builder /usr/local/bundle /usr/local/bundle
 ---> bc1ed877b946
Step 13/16 : COPY --from=builder /app /app
 ---> dc177f34bab8
Step 14/16 : WORKDIR /app
 ---> c54c7a2cfccf
Removing intermediate container 0bcf8fbb3080
Step 15/16 : RUN bundle exec rake assets:precompile
 ---> Running in bae031c2be1b
rake aborted!
LoadError: libpq.so.5: cannot open shared object file: No such file or directory - /usr/local/bundle/gems/pg-0.21.0/lib/pg_ext.so
/usr/local/bundle/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
/usr/local/bundle/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `block in require'
/usr/local/bundle/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:258:in `load_dependency'
/usr/local/bundle/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
/usr/local/bundle/gems/pg-0.21.0/lib/pg.rb:4:in `<top (required)>'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/runtime.rb:81:in `require'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/runtime.rb:81:in `block (2 levels) in require'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/runtime.rb:76:in `each'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/runtime.rb:76:in `block in require'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/runtime.rb:65:in `each'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/runtime.rb:65:in `require'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler.rb:114:in `require'
/app/config/application.rb:7:in `<top (required)>'
/app/Rakefile:5:in `require_relative'
/app/Rakefile:5:in `<top (required)>'
/usr/local/bundle/gems/rake-12.3.0/exe/rake:27:in `<top (required)>'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `load'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `kernel_load'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:28:in `run'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/cli.rb:424:in `exec'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/cli.rb:27:in `dispatch'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/cli.rb:18:in `start'
/usr/local/bundle/gems/bundler-1.16.1/exe/bundle:30:in `block in <top (required)>'
/usr/local/bundle/gems/bundler-1.16.1/lib/bundler/friendly_errors.rb:122:in `with_friendly_errors'
/usr/local/bundle/gems/bundler-1.16.1/exe/bundle:22:in `<top (required)>'
/usr/local/bundle/bin/bundle:104:in `load'
/usr/local/bundle/bin/bundle:104:in `<main>'
(See full trace by running task with --trace)
The command '/bin/sh -c bundle exec rake assets:precompile' returned a non-zero code: 1
```
</details>


This error is mysterying because the relevant directory should have been copied. In fact, the file it is looking for seems to be there in the intermediate container:

```
$ docker run -it -d c54c7a2cfccf /bin/bash
639c9ffa39c74b63eabfd777cf07d3d835cad4b831bcc6a33961b1ae613aa613
$ docker exec -it 639c9ffa39c74b63eabfd777cf07d3d835cad4b831bcc6a33961b1ae613aa613 /bin/bash
root@639c9ffa39c7:/app# ls -alF /usr/local/bundle/gems/pg-0.21.0/lib/pg_ext.so
-rwxr-xr-x 1 root root 959736 Jan 20 16:38 /usr/local/bundle/gems/pg-0.21.0/lib/pg_ext.so*

**Comment by Mr0grog** [comment]
Oh no, I’m so sorry I didn’t look at this earlier. I’m pretty certain the issue you were hitting here is that each stage in a multistage build is basically a totally independent image. When you copy files, you *only* get those files. So in the second stage, you’re missing all the packages you installed with `apt`. Instead of `FROM ruby:2.4.1-slim` for the second stage here, you would want `FROM base`.

**Comment by Mr0grog** [comment]
OK, I went ahead and rebased this on current master and fixed the Dockerfiles.

What I wound up doing was combining the import worker and server definitions into a single Dockerfile.

Having both in one file allows us to reduce duplication (they both share the same dependencies and source) without also adding some extra scripting around builds (otherwise we’d need three Dockerfiles and a script to make sure the base image is built before building the server or worker). This is less efficient than it could be, though — Docker builds every stage in the file, but then only keeps the one you asked for. Ideally we *would* have those three docker files + a shell script, make, or gulp.

How does this seem to you, @danielballan?

**Comment by danielballan** [comment]
Ah, that makes sense. Thanks for helping push this across the finish line.

**Comment by Mr0grog** [comment]
Ok, going to go ahead and merge this.

**Pull request merged**